docs/source/keys.rst

```diff
@@ -0,0 +1,293 @@
+For instance:
+
+.. code-block:: btm
+		
+    // Alice's public key
+    const kApub = pubkey:033806fe039c8d149c8e68f4665b4a479acab773b20bddf7df0df59ba4f0567341
+
+    // tx redeemable with Alice's private key
+    transaction A_funds {input = _ output = 1BTC: fun(x). versig(kApub; x)}
+
+
+If in the output script there is only one signature to 
+check, one can also use the address to specify the user, and 
+the behavour is the same.
+
+.. code-block:: btm
+		    
+    // Alice's address
+    const addrA = address:my6NmTELHBMVUsAWb34iRoGYDQpcYJvVZV
+
+    // tx redeemable with Alice's signature
+    transaction T {input = _ output = 1BTC: fun(x). versig(addrA; x)}
+    
+
+However, in case the signatures to be checked are more than one, the
+address cannot be used, and one must ressort to public keys. 
+(this restriction is due to Bitcoin transaction
+format). 
+Here is an example of the joint  verification of two signatures:
+
+.. code-block:: btm
+		
+	// Alice's public key	
+	const kApub = pubkey:03ff41f23b70b1c83b01914eb223d7a97a6c2b24e9a9ef2762bf25ed1c1b83c9c3
+	// Bob's public key
+	const kBpub = pubkey:03a5aded4cfa04cb4b49d4b19fe8fac0b58802983018cdd895a28b643e7510c1fb
+
+	// tx redeemable with two signatures: Alice's and Bob's
+	transaction T {
+		input = _
+		output = 1BTC: fun(x, y). versig(kApub, kBpub; x, y)
+	}
+
+In this case,  ``versig(kApub, kBpub; x, y)`` evaluates to ``true`` 
+if signature ``x`` has been made by Alice and signature ``y`` has been
+made by Bob.
+
+Notice that the order in which parameters are passed to ``versig``
+matters.  More generally, ``versig`` takes a lists of keys ``lk`` and
+a list of expressions ``le``.  Then, it  tries to
+verify the last signature in ``le``  with the last key in ``lk``.
+If they match, the function  proceeds to verify the previous signature in the
+list, otherwise it tries to verify the signature with the previous
+key.
+
+
+Thanks to this behaviour, it is possible to use ``versig`` to model
+more subtle conditions, such for instance, a *2-of-3* multi signature
+schema. For instance:
+
+.. code-block:: btm
+
+        // Alice's public key
+	const kApub = pubkey:03ff41f23b70b1c83b01914eb223d7a97a6c2b24e9a9ef2762bf25ed1c1b83c9c3
+	// Bob's public key
+	const kBpub = pubkey:03a5aded4cfa04cb4b49d4b19fe8fac0b58802983018cdd895a28b643e7510c1fb
+	//Carl's public key
+	const kCpub = pubkey:03bd94ee8e570da8815f5660bab86aca010d950ddfb87458bb0dcafbc8ea6f9657
+
+	//versig used to model  a *2-of-3* multi signature schema
+	transaction T {
+		input = _
+		output = 1BTC: fun(x, y). versig(kApub, kBpub, kCpub; x, y)
+	}
+
+
+In this case, output script ``versig(kApub, kBpub, kCpub; x, y)`` evaluates to true
+if the two parameters can match two of the three required keys.
+For instance, let ``sigC`` be Carl's signature  and ``sigB`` Bob's signature, then
+``versig(kApub, kBpub, kCpub; sigB, sigC)`` evaluates to ``true`` but
+``versig(kApub, kBpub, kCpub; sigC, sigB)`` does not. 
+
```

Comment:
Eliminerei da "Thanks to this behaviour....", considerato che il commento sopra riporta la tabellina. con tutti i casi.